File: python/newnanmanager/models/town.py
"""Town-related data models."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field

from .common import PagedData
from .player import Player


class Town(BaseModel):
    """城镇实体."""

    id: int = Field(description="城镇ID")
    name: str = Field(description="城镇名称")
    level: int = Field(description="城镇等级")
    leader_id: Optional[int] = Field(default=None, description="城主ID")
    qq_group: Optional[str] = Field(default=None, description="QQ群号")
    description: Optional[str] = Field(default=None, description="城镇描述")
    created_at: datetime = Field(description="创建时间")
    updated_at: datetime = Field(description="更新时间")


class TownsListData(PagedData[Town]):
    """城镇列表数据."""

    towns: list[Town] = Field(default_factory=list, description="城镇列表")

    @property
    def items(self) -> list[Town]:
        """获取数据项列表."""
        return self.towns

    @items.setter
    def items(self, value: list[Town]) -> None:
        """设置数据项列表."""
        self.towns = value


class TownMembersData(BaseModel):
    """城镇成员数据."""

    members: list[Player] = Field(default_factory=list, description="成员列表")
    total: int = Field(description="总数量")
    page: int = Field(description="当前页码")
    page_size: int = Field(description="每页大小")


class TownDetailResponse(BaseModel):
    """城镇详细信息响应."""

    town: Town = Field(description="城镇基本信息")
    leader: Optional[int] = Field(
        default=None, description="镇长ID"
    )  # 符合IDL中的 optional i32 leader 定义
    members: list[Player] = Field(default_factory=list, description="成员列表")

File: csharp/Models/Town.cs
using System.Text.Json.Serialization;

namespace NewNanManager.Client.Models;

/// <summary>
/// 城镇实体
/// </summary>
public class Town
{
    /// <summary>
    /// 城镇ID
    /// </summary>
    [JsonPropertyName("id")]
    public int Id { get; set; }

    /// <summary>
    /// 城镇名称
    /// </summary>
    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// 城镇等级
    /// </summary>
    [JsonPropertyName("level")]
    public int Level { get; set; }

    /// <summary>
    /// 城主ID
    /// </summary>
    [JsonPropertyName("leader_id")]
    public int? LeaderId { get; set; }

    /// <summary>
    /// QQ群号
    /// </summary>
    [JsonPropertyName("qq_group")]
    public string? QQGroup { get; set; }

    /// <summary>
    /// 城镇描述
    /// </summary>
    [JsonPropertyName("description")]
    public string? Description { get; set; }

    /// <summary>
    /// 创建时间
    /// </summary>
    [JsonPropertyName("created_at")]
    public DateTime CreatedAt { get; set; }

    /// <summary>
    /// 更新时间
    /// </summary>
    [JsonPropertyName("updated_at")]
    public DateTime UpdatedAt { get; set; }
}

/// <summary>
/// 城镇列表数据
/// </summary>
public class TownsListData : PagedData<Town>
{
    /// <summary>
    /// 城镇列表
    /// </summary>
    [JsonPropertyName("towns")]
    public List<Town> Towns
    {
        get => Items;
        set => Items = value;
    }
}

/// <summary>
/// 城镇成员数据
/// </summary>
public class TownMembersData : PagedData<Player>
{
    /// <summary>
    /// 成员列表
    /// </summary>
    [JsonPropertyName("members")]
    public List<Player> Members
    {
        get => Items;
        set => Items = value;
    }
}

/// <summary>
/// 城镇详细信息响应
/// </summary>
public class TownDetailResponse
{
    /// <summary>
    /// 城镇基本信息
    /// </summary>
    [JsonPropertyName("town")]
    public Town Town { get; set; } = new();

    /// <summary>
    /// 镇长ID
    /// </summary>
    [JsonPropertyName("leader")]
    public int? Leader { get; set; } // 符合IDL中的 optional i32 leader 定义

    /// <summary>
    /// 成员列表（包括镇长）
    /// </summary>
    [JsonPropertyName("members")]
    public List<Player> Members { get; set; } = new();
}
